### DISCUSSION TITLE
Help with examples of grouping routes with go-chi

### DISCUSSION THREAD

**Discussion by lays147** [danielgtaylor/huma#589] [Q&A]
Hi, thanks for this amazing framework. 
However, I'm having difficulties on implementing some configurations since all the examples are based on all code within the `main.go` file, and as we know, in the real world things doesn't work like that. 

Could you help me with the following scenario?

I have a group of resources that will be available under the `resources/v1/` path. And other group that will be under the `resources/v2/` path. 
Based on my interpretation of the docs, every time that I register a new operation I'll need to append those full paths to the resources like:
```go
	huma.Register(api, huma.Operation{
		OperationID:   "new-receivable",
		Method:        http.MethodPost,
		Path:          "/resources/v1/receivables",
		Summary:       "Create a new receivable",
		Description:   "Request eh creation of a new receivable",
		DefaultStatus: http.StatusCreated,
	}, func())
```

But I would like to avoid this hard coded thing on the paths and add this prefix somewhere else, because this kind of configuration for a lot of routes can easily be misconfigured.
How can one configure these groups like the [go-chi](https://go-chi.io/#/pages/routing?id=routing-groups) docs show? 

Also, it would be wonderful to have a real world example on how to use Huma in a real world project where all the code is broken in other files instead of everything being on the main.

**Comment by lays147** [comment]
@danielgtaylor could you give any advice on how to achieve the proposed grouping of routes above?

**Reply by vthang95** [reply]
Hi, I have the same problem and  this is my setup. Maybe it can help:

```go
app.Server = chi.NewMux()
config := huma.DefaultConfig("App API", "1.0.0")
api := humachi.New(app.Server, config)
...
app.Server.Group(func(mux chi.Router) {
	handler.RegisterHeathcheck(mux, api)

	handler.RegisterAuth(mux, api)
})

app.Server.Group(func(mux chi.Router) {
	api.UseMiddleware(handler.GetAuthMiddleware(api))

	handler.RegisterAccount(mux, api)
})
```

```go
// healthcheck.go
package apisV1
...
func (h *Handler) RegisterHeathcheck(mux chi.Router, api huma.API) {
	tags := []string{"Healthcheck"}

	huma.Register(api, huma.Operation{
		OperationID: "ping",
		Method:      http.MethodGet,
		Path:        "/healthcheck/ping",
		Tags:        tags,
	}, h.getPing)
}

type pingOutput struct {
	Body struct {
		Message string `json:"message" example:"pong"`
	}
}

func (h *Handler) getPing(ctx context.Context, input *EmptyInput) (*pingOutput, error) {
	resp := &pingOutput{}
	resp.Body.Message = "pong!"
	return resp, nil
}
```

```go
// auth.go
package apisV1
...
func (h *Handler) RegisterAuth(mux chi.Router, api huma.API) {
	tags := []string{"Auth"}

	prefix := "/auth"

	huma.Register(api, huma.Operation{
		OperationID: "login",
		Method:      http.MethodPost,
		Path:        prefix + "/login",
		Tags:        tags,
	}, h.login)

	huma.Register(api, huma.Operation{
		OperationID: "signup",
		Method:      http.MethodPost,
		Path:        prefix + "/signup",
		Tags:        tags,
	}, h.signup)
}
```

There are some issues with using route prefixes, as they can lead to mismatches between the OpenAPI documentation and the actual routes. Therefore, I believe that separating the code into different files and placing the prefix at the top of each file is the best solution for me.

**Reply by dejayc** [reply]
@vthang95 You're passing in `mux` to `RegisterHealthcheck` and `RegisterAuth`, but as far as I can see, `mux` is not used in any way within those functions.  Therefore, passing in `mux` does nothing.

**Reply by vthang95** [reply]
```go
app.Server.Group(func(mux chi.Router) {
        adapter := humachi.NewAdapter(mux)
	api := huma.NewAPI(config, adapter)
	handler.RegisterHeathcheck(api)

	handler.RegisterAuth(api)
})
```
@dejayc Sorry, I didn’t read what I typed carefully.
